modules/core/04-channel/types/msgs.go

```diff
@@ -723,8 +723,8 @@ func (msg MsgChannelUpgradeOpen) ValidateBasic() error {
 		return errorsmod.Wrap(commitmenttypes.ErrInvalidProof, "cannot submit an empty channel proof")
 	}
 
-	if !collections.Contains(msg.CounterpartyChannelState, []State{TRYUPGRADE, ACKUPGRADE, OPEN}) {
-		return errorsmod.Wrapf(ErrInvalidChannelState, "expected channel state to be one of: %s, %s or %s, got: %s", TRYUPGRADE, ACKUPGRADE, OPEN, msg.CounterpartyChannelState)
+	if msg.CounterpartyChannelState != STATE_FLUSHCOMPLETE {
```

Comment:
note to reviewers: this is somewhat unrelated but we wanted to use the handlers in one of the test cases (counterparty in OPEN) and bumped into it.

Reply:
counterparty is allowed to be `OPEN` for `chanUpgradeOpen`

Reply:
good catch!